activerecord/lib/active_record/relation/predicate_builder.rb

```diff
@@ -28,9 +28,9 @@ def build_from_hash(attributes, &block)
     def self.references(attributes)
       attributes.each_with_object([]) do |(key, value), result|
         if value.is_a?(Hash)
-          result << Arel.sql(key)
+          result << Arel.sql(key, retryable: true)
```

Comment:
Not sure whether this (or below) can potentially contain an expression? (I actually also haven't checked whether this method's result ends up in a constructed query... I'm not immediately thinking of why it would. [Though the use of `Arel.sql` is clearly suggesting I'm wrong.])

Reply:
Hm, `attributes` come from a `where` clause, so probably we should not retry here to be safe.